When using sysroot we should append it to libdir which is helpful in cross builds
as the system is staged in the sysroot. For normal build i.e. when lt_sysroot is not
set it will still behave same and add -L/usr/lib to relink command

-Khem

Index: libtool-2.4/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.4.orig/libltdl/config/ltmain.m4sh
+++ libtool-2.4/libltdl/config/ltmain.m4sh
@@ -6122,7 +6122,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      #add_dir="-L$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
